From bfed380b7886ff4351c57bd021b25a8e98ddf461 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sat, 28 Sep 2024 14:54:36 +0200
Subject: [PATCH 04/22] lx2160acex7: rename sdhc1 config to generic sdhc, for
 both sd & emmc

Both sdhc1 and sdhc2 boot can share same rcw.
They use identical boot location pointer values, differences were only
in old rcw with explicit block-copy from boot media to ocram.

This block-copy is not required for either spi, emmc or sd boot.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_2400_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_2600_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_2900_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_3200_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_2400_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_2600_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_2900_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_3200_8_5_2_sdhc.rcw} | 0
 lx2160acex7/include/common.rcwi                           | 8 ++++++--
 lx2160acex7/include/common_pbi.rcwi                       | 2 +-
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_2400_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_2600_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_2900_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2000_700_3200_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_2400_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_2600_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_2900_8_5_2_sdhc.rcw} | 0
 ...0_8_5_2_sdhc1.rcw => rcw_2200_700_3200_8_5_2_sdhc.rcw} | 0
 18 files changed, 7 insertions(+), 3 deletions(-)
 rename lx2160acex7/clearfog-cx/{rcw_2000_700_2400_8_5_2_sdhc1.rcw => rcw_2000_700_2400_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2000_700_2600_8_5_2_sdhc1.rcw => rcw_2000_700_2600_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2000_700_2900_8_5_2_sdhc1.rcw => rcw_2000_700_2900_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2000_700_3200_8_5_2_sdhc1.rcw => rcw_2000_700_3200_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2200_700_2400_8_5_2_sdhc1.rcw => rcw_2200_700_2400_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2200_700_2600_8_5_2_sdhc1.rcw => rcw_2200_700_2600_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2200_700_2900_8_5_2_sdhc1.rcw => rcw_2200_700_2900_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7/clearfog-cx/{rcw_2200_700_3200_8_5_2_sdhc1.rcw => rcw_2200_700_3200_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2000_700_2400_8_5_2_sdhc1.rcw => rcw_2000_700_2400_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2000_700_2600_8_5_2_sdhc1.rcw => rcw_2000_700_2600_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2000_700_2900_8_5_2_sdhc1.rcw => rcw_2000_700_2900_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2000_700_3200_8_5_2_sdhc1.rcw => rcw_2000_700_3200_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2200_700_2400_8_5_2_sdhc1.rcw => rcw_2200_700_2400_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2200_700_2600_8_5_2_sdhc1.rcw => rcw_2200_700_2600_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2200_700_2900_8_5_2_sdhc1.rcw => rcw_2200_700_2900_8_5_2_sdhc.rcw} (100%)
 rename lx2160acex7_rev2/clearfog-cx/{rcw_2200_700_3200_8_5_2_sdhc1.rcw => rcw_2200_700_3200_8_5_2_sdhc.rcw} (100%)

diff --git a/lx2160acex7/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc1.rcw b/lx2160acex7/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc1.rcw
rename to lx2160acex7/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc.rcw
diff --git a/lx2160acex7/include/common.rcwi b/lx2160acex7/include/common.rcwi
index 5523318..c76b174 100644
--- a/lx2160acex7/include/common.rcwi
+++ b/lx2160acex7/include/common.rcwi
@@ -11,8 +11,12 @@ C8_PLL_SEL=0
 HWA_CGA_M1_CLK_SEL=1
 /* Cluster group B clock is PLL2 div 2 (for DCE) */
 HWA_CGB_M1_CLK_SEL=6
-/* fall-back boot-mode is ocram, when DCFG bit location pointer registers are null */
-BOOT_LOC=21 // TODO: test if SPI boot still functional
+/*
+ * fall-back boot-mode when DCFG boot location pointer registers are null
+ * - 0b10101 (21): OCRAM
+ * - 0b11010 (26): XSPI
+ */
+BOOT_LOC=21
 /* SYSCLK is 100MHz */
 SYSCLK_FREQ=600
 /* USB-3.0 clock is 100MHz */
diff --git a/lx2160acex7/include/common_pbi.rcwi b/lx2160acex7/include/common_pbi.rcwi
index 848d48b..36b723d 100644
--- a/lx2160acex7/include/common_pbi.rcwi
+++ b/lx2160acex7/include/common_pbi.rcwi
@@ -10,7 +10,7 @@ write 0x2320000,0x20000000
 /* Errata to write on scratch reg for validation */
 #include <../lx2160asi/scratchrw1.rcw>
 
-/* Boot Location Pointer */
+/* Set Boot Location Pointer (Fall-back when unset is BOOT_LOC) */
 #if defined(LX_BOOTSOURCE_SDHC1)
 #include <../lx2160asi/bootlocptr_sd.rcw>
 #elif defined(LX_BOOTSOURCE_XSPI)
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2400_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2600_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2000_700_2900_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2000_700_3200_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2400_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2600_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2200_700_2900_8_5_2_sdhc.rcw
diff --git a/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc1.rcw b/lx2160acex7_rev2/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc.rcw
similarity index 100%
rename from lx2160acex7_rev2/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc1.rcw
rename to lx2160acex7_rev2/clearfog-cx/rcw_2200_700_3200_8_5_2_sdhc.rcw
-- 
2.43.0

